packages/phishing-controller/CHANGELOG.md

```diff
@@ -7,6 +7,31 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [12.0.0]
+
+### Added
+
+- Define and export new types: `PhishingControllerGetStateAction`, `PhishingControllerStateChangeEvent`, `PhishingControllerEvents` ([#4633](https://github.com/MetaMask/core/pull/4633))
+- add allowlist to c2 domain detection ([#4464](https://github.com/MetaMask/core/pull/4644))
+- add Block Client Side C2 Requests by Managing a hashed C2 Request Blocklist
+  ([#4526](https://github.com/MetaMask/core/pull/4526))
+- add requestBlocklist type to ListTypes.
+- add isBlockedRequest method to PhishingController.
+- add Logic to update and check requestBlocklist in #updateStalelist.
+- add isMaliciousRequestDomain method to PhishingDetector.
+- add Handling of requestBlocklist in PhishingDetector configuration.
+- add sha256 and toHex imports from ethereum-cryptography.
+- add sha256Hash function to generate SHA-256 hash of a domain.
+
+### Changed
+
+- **BREAKING:** Added @noble/hashes dependency version ^1.4.0.
+- **BREAKING:**: Added ethereum-cryptography dependency version ^2.1.2.
+- **BREAKING:** `PhishingControllerMessenger` must allow internal events defined in the `PhishingControllerEvents` type ([#4633](https://github.com/MetaMask/core/pull/4633))
+- `PhishingControllerActions` is widened to include the `PhishingController:getState` action ([#4633](https://github.com/MetaMask/core/pull/4633))
+- Bump `@metamask/base-controller` from `^6.0.2` to `^6.0.3` ([#4625](https://github.com/MetaMask/core/pull/4625))
+- Bump `@metamask/controller-utils` from `^11.0.2` to `^11.1.0` ([#4639](https://github.com/MetaMask/core/pull/4639))
```

Comment:
Wording, format, order (we usually put more important entries first -- feel free to reorder):

```suggestion
### Added

- Add allowlist functionality to the C2 domain detection system ([#4464](https://github.com/MetaMask/core/pull/4644))
- Add `PhishingController` functionality for blocking client-side C2 requests by managing a hashed C2 request blocklist ([#4526](https://github.com/MetaMask/core/pull/4526))
  - Add `requestBlocklist` type to `ListTypes`.
  - Add `isBlockedRequest` method to `PhishingController`.
  - Add `isMaliciousRequestDomain` method to `PhishingDetector`.
  - Add handling of `requestBlocklist` in `PhishingDetector` configuration.
  - Add logic to update and check `requestBlocklist` when updating a stale list.
  - Add `sha256Hash` function to generate SHA-256 hash of a domain.
- Define and export new types: `PhishingControllerGetStateAction`, `PhishingControllerStateChangeEvent`, `PhishingControllerEvents` ([#4633](https://github.com/MetaMask/core/pull/4633))

### Changed

- **BREAKING:** Add `@noble/hashes` `^1.4.0` as dependency ([#4526](https://github.com/MetaMask/core/pull/4526))
- **BREAKING:**: Add `ethereum-cryptography` `^2.1.2` as dependency ([#4526](https://github.com/MetaMask/core/pull/4526))
- **BREAKING:** `PhishingControllerMessenger` must allow internal events defined in the `PhishingControllerEvents` type ([#4633](https://github.com/MetaMask/core/pull/4633))
- Widen `PhishingControllerActions` to include the `PhishingController:getState` action ([#4633](https://github.com/MetaMask/core/pull/4633))
- Bump `@metamask/base-controller` from `^6.0.2` to `^6.0.3` ([#4625](https://github.com/MetaMask/core/pull/4625))
- Bump `@metamask/controller-utils` from `^11.0.2` to `^11.1.0` ([#4639](https://github.com/MetaMask/core/pull/4639))
```